include/NovelRT/Ecs/DefaultComponentTypes.h

```diff
@@ -0,0 +1,85 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+
+#ifndef NOVELRT_DEFAULTCOMPONENTTYPES_H
+#define NOVELRT_DEFAULTCOMPONENTTYPES_H
+
+#ifndef NOVELRT_ECS_H
+#error NovelRT does not support including types explicitly by default. Please include Ecs.h instead for the Ecs namespace subset.
+#endif
+
+namespace NovelRT::Ecs
+{
+    struct TransformComponent
+    {
+        Maths::GeoVector3F positionAndLayer = Maths::GeoVector3F::zero();
+        Maths::GeoVector2F scale = Maths::GeoVector2F::one();
+        float rotationInEulerAngles = 0.0f;
+
+        inline TransformComponent& operator+=(const TransformComponent& other)
+        {
+            positionAndLayer += other.positionAndLayer;
+            rotationInEulerAngles += other.rotationInEulerAngles;
+            scale += other.scale;
+
+            if (rotationInEulerAngles > 359.999999f) // 360 == 0 pretty sure :]
+            {
+                rotationInEulerAngles -= 359.999999f;
+            }
+            else if (rotationInEulerAngles < 0)
```

Comment:
You might consider allowing this to be up to `-360.0f`. Even though the ultimate position is the "same", the "direction" is different and may impact interpolation or blur

Reply:
To clarify - what would be the exact behaviour of  `-360`? Would it wrap around to `0` or go `+360`?

Reply:
Just to elaborate a bit on this one; consider, for example, a clock face.

Winding forward by 30 degrees puts it at 1, while winding it backwards by 330 degrees puts it at the same position.

However, if the rotation is `30 degrees` and the time to transition is `10 s` then at 5s the hand will be halfway between 12 and 1. Where-as for `-330` degrees it should have been somewhere between the 6 and 7 markers.

Reply:
@tannergooding I'll need you to double check I've implemented this one correctly before marking as resolved.